ruma-events/README.md

```diff
@@ -13,3 +13,193 @@ ruma-events requires Rust 1.40.0 or later.
 ## Documentation
 
 ruma-events has [comprehensive documentation](https://docs.rs/ruma-events) available on docs.rs.
+
+# How-To
+
+## Deserializing
+
+Ruma offers the `EventJson` wrapper to enable passing around JSON text that is semi validated. This
+is useful when [extending ruma](TODO_link). All event structs and enums implement `Serialize/Deserialize` so
+EventJson is not strictly needed anymore (pre 0.22 this was not the case).
+
+```rust
+let json = r#"{ "type": "imagine a full event", "content": {...} }"#;
+
+let deser = serde_json::from_str::<EventJson<AnyRoomEvent>>()
+    .unwrap() // the first Result from serde_json::from_str
+    .deserialize() // this deserializes the event while still keeping the RawJson in EventJson
+    .unwrap() // finally get to the AnyRoomEvent
+
+// event content has a trait function that allows for validated deserialization
+let content = r#"{ "membership": "imagine member event content", ... }"#;
+let deser = EventContent::from_parts("m.room.member", serde_json::value::to_raw_value(content)).unwrap();
+// There is also a method to convert the content to the redacted version according to the Matrix spec
+// The only field left is `membership`
+let redacted = deser.redact();
+```
+
+## Creating Events to Extend Ruma
```

Comment:
I love this section! ♥️

Reply:
If you're talking about the `Creating Events to Extend Ruma` I know it's AWSOME how little code it takes to add your own event :partying_face:! Really good API we (you mostly) have come up with.